File: radar-commons-server/src/main/java/org/radarcns/util/SpecificAvroConvertible.java
package org.radarcns.util;

import org.apache.avro.specific.SpecificRecord;

/**
 * Convertible from Avro. This assumes that the entire state of an object can be replaced
 * with the data in the record, using e.g. an empty constructor.
 */
public interface SpecificAvroConvertible {
    /** Converts the convertible to a specificrecord. */
    SpecificRecord toAvro();

    /** Reads the convertible from a specificrecord. */
    void fromAvro(SpecificRecord record);
}

File: radar-commons-server/src/main/java/org/radarbase/util/SpecificAvroConvertible.kt
package org.radarbase.util

import org.apache.avro.specific.SpecificRecord

/**
 * Convertible from Avro. This assumes that the entire state of an object can be replaced
 * with the data in the record, using e.g. an empty constructor.
 */
interface SpecificAvroConvertible {
    /** Converts the convertible to a specificrecord.  */
    fun toAvro(): SpecificRecord

    /** Reads the convertible from a specificrecord.  */
    fun fromAvro(record: SpecificRecord)
}
